tmt/steps/report/reportportal.py

```diff
@@ -38,113 +25,179 @@ class ReportReportPortalData(tmt.steps.report.ReportStepData):
         option="--token",
         metavar="TOKEN",
         default=os.environ.get("TMT_REPORT_REPORTPORTAL_TOKEN"),
-        help="The token to use for upload to the ReportPortal instance.")
-    project: Optional[str] = tmt.utils.field(
+        help="The token to use for upload to the ReportPortal instance (from RP user profile).")
+    project: Optional[str] = tmt.utils.field(  # todo: argument should be mandatory
         option="--project",
-        metavar="PROJECT",
+        metavar="PROJECT_NAME",
         default=None,
         help="The project name which is used to create the full URL.")
-    launch_name: Optional[str] = tmt.utils.field(
-        option="--launch-name",
-        metavar="NAME",
+    launch: Optional[str] = tmt.utils.field(
+        option="--launch",
+        metavar="LAUNCH_NAME",
+        default=None,
+        help="The launch name (name of plan per launch is used by default).")
+    env_var: Optional[str] = tmt.utils.field(
+        option="--env-var",
+        metavar="TMT_VAR",
         default=None,
-        help="The launch name (base name of run id used by default).")
+        multiple=True,
+        help="List one of the environment variables that begin with \"TMT_\" to disobey \
+the rule of not displaying them in Report Portal launch (the argument can be repeated).")
 
 
 @tmt.steps.provides_method("reportportal")
 class ReportReportPortal(tmt.steps.report.ReportPlugin):
     """
-    Report test results to a ReportPortal instance
+    Report test results to a ReportPortal instance via reportportal_client API
 
     Requires a TOKEN for authentication, a URL of the ReportPortal
-    instance and the PROJECT name. In addition to command line options
-    it's possible to use environment variables to set the url and token:
+     instance and the PROJECT name. In addition to command line options
+     it's possible to use environment variables to set the url and token:
 
         export TMT_REPORT_REPORTPORTAL_URL=...
         export TMT_REPORT_REPORTPORTAL_TOKEN=...
 
-    The optional launch NAME is passed to ReportPortal. Assuming the URL
-    and TOKEN variables are provided by the environment, the config can
-    look like this:
+    The optional launch NAME doesn't have to be provided if it is the same
+     as the plan name (by default). Assuming the URL and TOKEN variables
+     are provided by the environment, the config can
+     look like this:
 
         report:
             how: reportportal
             project: baseosqe
-            launch-name: maven
+
+    In item details of the ReportPortal instance, the environment variables
```

Comment:
It is not clear for me

As far as i can see the test properties, exist in test-> ItemDetails? 
What does this parameter do?

Reply:
What parameter do you mean?

Reply:
```
per test are displayed except those that begin with ``TMT_``
```
Does it mean that all env vars starts with TMT_ will not be upload to properties of tests case?

Reply:
I do not understand the usage of 
--env-vars: [TMT_TEST_DATA,TMT_PLAN_DATA] this, in case it is not displayed 
Where does the env vars displayed at the end?
I tried it, and did not see it in the properties or the attributes

Reply:
I tried to sum up it shortly there.

Considering that this is probably a rare use case, in order to make it less confusing,there is a thought to remove this argument entirely and move this option only to env variable (e.g. export TMT_REPORT_REPORTPORTAL_TMT_ENV_VARS=...).
Need more opinions/insights here.

Reply:
Thanks for explaining it,
I do not see the usage of properties, per now, important.
Most interesting per me is the attributes metadata, since it is filtered,
The filter is required for dashboards and statistics